core/src/main/kotlin/org/evomaster/core/search/gene/DateTimeGene.kt

```diff
@@ -44,9 +53,22 @@ open class DateTimeGene(
     }
 
     override fun getValueAsRawString(): String {
-        return "${date.getValueAsRawString()}" +
-                "T" +
-                "${time.getValueAsRawString()}"
+        val formattedDate = GeneUtils.let {
+            "${it.padded(date.year.value, 4)}-${it.padded(date.month.value, 2)}-${it.padded(date.day.value, 2)}"
+        }
+        val formattedTime = GeneUtils.let {
+            "${it.padded(time.hour.value, 2)}:${it.padded(time.minute.value, 2)}:${it.padded(time.second.value, 2)}"
+        }
+        return when (dateTimeGeneFormat) {
+            DateTimeGeneFormat.ISO_LOCAL_DATE_TIME_FORMAT -> {
+                "${formattedDate}" + "T" + "${formattedTime}"
+            }
```

Comment:
why not simplifying it with `"${formattedDate}T${formattedTime}"`?

Reply:
agree